src/main/java/gift/repository/ProductRepositoryImpl.java

```diff
@@ -0,0 +1,62 @@
+package gift.repository;
+
+import gift.entity.Product;
+import org.springframework.stereotype.Repository;
+
+import java.util.HashMap;
+import java.util.List;
+
+@Repository
+public class ProductRepositoryImpl implements ProductRepository {
+
+    // 컬렉션을 활용하여 임시로 구현
+    final private HashMap<Long, Product> productMap;
+    private Long sequence; // ID 생성에 사용될 시퀀스
+
+    public ProductRepositoryImpl() {
+        this.productMap = new HashMap<>();
+        this.sequence = 0L;
+
+        testInit();
+    }
+
+    // 테스트 용으로 초기 데이터를 설정
+    private void testInit() {
+        productMap.put(8146027L,
+                new Product(8146027L, "Gift Card", 10000L, "https://example.com/giftcard.jpg"));
+    }
+
+    @Override
+    public List<Product> findAll() {
+        return productMap.values().stream().toList();
+    }
+
+    @Override
+    public Product findById(Long productId) {
+        return productMap.get(productId);
+    }
+
+    @Override
+    public Product save(Product product) {
+        if (product.getId() == null) {
+            product.setId(++sequence); // ID가 없으면 시퀀스를 증가시켜 새 ID를 할당
+        }
+        productMap.put(product.getId(), product); // 제품을 저장
+        return product; // 저장된 제품 반환
+    }
+
+    @Override
+    public Product update(Product product) {
+        productMap.put(product.getId(), product); // 제품을 업데이트
+        return product; // 업데이트된 제품 반환
+    }
+
+    @Override
+    public Long deleteById(Long productId) {
+        if (productMap.containsKey(productId)) {
+            productMap.remove(productId); // 제품을 삭제
+            return 1L; // 영향을 받은 행 수 반환
```

Comment:
deleteById를 인터페이스에서 봤을 때, 반환 값이 삭제된 ID가 리턴되는 줄 알았는데 그게 아니었군요 ㅎㅎ
영향을 받은 행 수를 반환하는 거면 Long 보단 boolean이 좋을 수도 있을 것 같아요.
```suggestion
@Override
public boolean deleteById(Long productId) {
    return productMap.remove(productId) != null;
}
```